src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs

```diff
@@ -397,28 +388,17 @@ public static void CreateAssociation(object primary, object secondary)
                 throw new ArgumentException(SR.TypeDescriptorSameAssociation);
             }
 
-            if (s_associationTable == null)
-            {
-                lock (s_internalSyncObject)
-                {
-                    if (s_associationTable == null)
-                    {
-                        s_associationTable = new WeakHashtable();
-                    }
-                }
-            }
-
-            IList associations = (IList)s_associationTable[primary];
+            IList associations = (IList)s_associationTableLazy.Value[primary];
 
             if (associations == null)
             {
-                lock (s_associationTable)
+                lock (s_associationTableLazy)
```

Comment:
This is changing the semantic, in that previously it was locking on the table, now it's locking on the lazy. Is there code elsewhere that was expecting all locking to be done on the table itself?  Let's keep it the way it was, e.g. lock (s_associationTableLazy.Value) rather than lock (s_associationTableLazy)